shared/subscribers/service.go

```diff
@@ -82,16 +97,30 @@ func (s *Service) ObserveEvents(ctx context.Context, retries int, eTypes []event
 			case bs := <-in:
 				// batch size changed: drain buffer and send data
 				data = append(data, sub.UpdateBatchSize(ctx, bs)...)
-				if len(data) > 0 {
+				dataLength := len(data)
+
+				if dataLength > 0 {
 					trySend()
 				}
+
+				if dataLength > s.maxBufferSize {
+					s.log.Warningf("slow consumer detected, closing event observer")
+					return
+				}
 			default:
 				// wait for actual changes
 				data = append(data, sub.GetData(ctx)...)
+				dataLength := len(data)
+
 				// this is a very rare thing, but it can happen
-				if len(data) > 0 {
+				if dataLength > 0 {
 					trySend()
 				}
+
+				if dataLength > s.maxBufferSize {
```

Comment:
Same here

Reply:
if it can send the data then by definition its not a slow consumer, so checking before doesn't make sense

Reply:
switched them